test_runner/batch_others/test_proxy.py

```diff
@@ -22,6 +29,154 @@ def test_password_hack(static_proxy):
         static_proxy.safe_psql('select 1', sslsni=0, user=user, password=magic)
 
 
+async def get_session_id_from_welcome_message(local_link_proxy, proc):
```

Comment:
Is `local_link_proxy` used? What does `proc` mean? The function says `get_session_id_from_welcome_message`, so I expect it to take in the welcome message string. IMO would be a lot clearer to:
1. Read all lines from stderr
2. Pass the lines to this function
3. Find the one line that contains `dymmy-uri` (there should be exactly one)
4. Parse the session id from it

Reply:
local_link_proxy is used, proc is the process that communicates with the local_link_proxy.

okay will do!